src/app/controllers/LoginController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\controllers;
+
+use App\Core\Controller;
+use App\core\Responses\HTMLResponse;
+use App\core\Responses\IResponse;
+use App\models\LoginModel;
+
+class LoginController extends Controller
+{
+    public function login($params): IResponse
+    {
+        $data = [...$params];
+
+        if ((isset($_SESSION['user'], $_SESSION['email']))) {
+            $data['user'] = ['email' => $_SESSION['email'], 'name' => $_SESSION['user']];
+        }
+
+        $body = $this->view->render('login', $data);
+
+        return new HTMLResponse(['200 OK'], $body);
+    }
+
+    public function loginPost($params): IResponse
+    {
+        $data = [];
+
+        if (!isset($params['email'], $params['username'], $params['password'])) {
+            $data['error'] = 'Please fill in all fields';
+            $body = $this->view->render('login', $data);
+
+            return new HTMLResponse(['200 OK'], $body);
+        }
+
+        $model = new LoginModel($params['email'], $params['username'], $params['password']);
+
+        $data['errors'] = $model->validate();
+
+        if (!empty($data['errors'])) {
+            return new HTMLResponse(['400 Bad Request'], $this->view->render('login', $data));
+        }
+
+        $foundedUser = $this->model->fetch($model->email);
+
+        if ($foundedUser === null) {
```

Comment:
Could: Security breach. Error messages are different for situation whether user not found and wrong password